lib/ledger_sync/ledgers/xero/resources/account.rb

```diff
@@ -0,0 +1,45 @@
+
+# frozen_string_literal: true
+
+module LedgerSync
+  module Ledgers
+    module Xero
+      class Account < Xero::Resource
+
+        TYPES = {
+          'bank' => 'BANK',
+          'current' => 'CURRENT',
+          'current_liability' => 'CURRLIAB',
+          'depreciation' => 'DEPRECIATN',
+          'direct_costs' => 'DIRECTCOSTS',
+          'equity' => 'EQUITY',
+          'expense' => 'EXPENSE',
+          'fixed_asset' => 'FIXED',
+          'inventory' => 'INVENTORY',
+          'liability'=> 'LIABILITY',
+          'non_current' => 'NONCURRENT',
+          'other_income' => 'OTHERINCOME',
+          'overheads' => 'OVERHEADS',
+          'pre_payment' => 'PREPAYMENT',
+          'revenue' => 'REVENUE',
+          'sales' => 'SALES',
+          'long_term_liability' => 'TERMLIAB',
+          'pay_as_you_go_liability' => 'PAYGLIABILITY',
+          'pay_as_you_go' => 'PAYG',
+          'superannuation_expense' => 'SUPERANNUATIONEXPENSE',
+          'superannuation_liability' => 'SUPERANNUATIONLIABILITY',
+          'wages_expense' => 'WAGESEXPENSE',
+        }.freeze
+
+        attribute :code, type: LedgerSync::Type::String
+        attribute :name, type: LedgerSync::Type::String
+        attribute :account_id, type: LedgerSync::Type::String
+        attribute :type, type: Type::StringFromSet.new(values: TYPES.keys)
+
+        def name
```

Comment:
This is convention that we should probably break from. We started using `name` method for every resource as kind of identifier. Which is why we pretty much just alias `name` to `Name` attribute method.

It was nice to be able to call `resource.name` and get back object readable identifier (ie Order number, ...). But I would honestly clean all these away.

Anyways, as your attribute is lowecare `name` on line 35, it would create reader method for it. And on line 40 youre expecting to have reader method with capital `Name`. We match attributes to how they are represented in ledgers API, so if they use `Name` convention, our attribute should be called `Name` as well.

Reply:
I believe our convention of having the attribute names match that of the API supercedes having the `.name` method for resources. So if the API uses `name` it would be a resource attribute. That said, this could be done if we move all resource attributes under `resource.attributes.name` where `resource.attributes` is the attribute set. I'm still OK making this transition as it frees up the resource namespace for helpful resource methods.